elrond_dex_farm/src/lib.rs

```diff
@@ -202,30 +265,59 @@ pub trait Farm:
             total_entering_amount: entering_amount.clone(),
             total_liquidity_amount: re_added_liquidity.clone(),
             entering_epoch: farm_attributes.entering_epoch,
+            with_locked_rewards: farm_attributes.with_locked_rewards,
         };
         let new_nonce =
             self.create_farm_tokens(&re_added_liquidity, &farm_token_id, &new_attributes);
         self.send()
             .transfer_tokens(&farm_token_id, new_nonce, &re_added_liquidity, &caller);
 
         // Send rewards
-        self.send()
-            .transfer_tokens(&reward_token_id, 0, &reward, &caller);
+        let mut reward_nonce = 0u64;
+        if farm_attributes.with_locked_rewards {
+            if reward > 0 {
```

Comment:
add a require instead of the IF.

Reply:
a require would end the execution and fail the transaction whereas in case of 0 rewards, we still need to send the LP tokens (for example) back

Reply:
refactored to be more intuitive